### DISCUSSION TITLE
"Show persistent overlay" is not persistent always because it does not show state of more than one key

### DISCUSSION THREAD

**Discussion by KodeXall** [jonaskohl/CapsLockIndicator#131] [Ideas]
Under the

**Notification > Display time > Show persistent overlay when...**

a user can adjust Notifications to individual needs. Unfortunately when a user has the persistent overlay set only for e.g. CapsLock, but under the

**General > show notification when...**

all three keys are selected, then after pressing of CapsLock [thus setting it in the on / down state] followed by pressing of NumLock or ScrollLock, that supposed persistent overlay not only disappears but also does not come back when this temporary one for NumLock or ScrollLock fades away

And so Notifications should either show up one above another [thus creating on a screen a short column with up to the 3 rows] - or be presented just like now in the same place overriding each other but then coming back to the last one active which happened to be customized as a persistent. However that second solution would still lead to not fully informative situations, as the user can have all 3 keys set as persistent ones, right? Plus users already can have 3 separate Icons in the Tray - so why not 3 concurrent Notifications? I tried going to

**Advanced Options > Advanced Settings**

but did not find ones leading to a such configuration

If 3 always persistent Notifications could be setup at the same time it would be potentially extra beneficial for a user to be able to choose also their position in relation to each other and adjust their sizes. I for example would set a big Notification for CapsLock and below or above it the 2 remaining ones, as they are less important to me [each of those 2 with half the width of the main one, for aesthetic reasons]. And if they positions would be fixed, I would not have to read them, because from the mare shape created by the adjacent Notifications I could tell was is or just was pressed. And if every Notification could have its own colors, the for I would test out if it would not be better [in terms of immediate distinction] to have one color for all 3 in the ON state but three different for OFF

**Comment by jonaskohl** [comment]
A revamp of the notification system is long overdue and already on my backlog
